src/custom/utils/operator.ts

```diff
@@ -0,0 +1,131 @@
+
+export interface OrderPostError {
+  errorType: 'MissingOrderData' | 'InvalidSignature' | 'DuplicateOrder' | 'InsufficientFunds'
+  description: string
+}
+
+function _getApiBaseUrl(chainId: ChainId): string {
+  const baseUrl = API_BASE_URL[chainId]
+
+  if (!baseUrl) {
+    throw new Error('Unsupported Network. The operator API is not deployed in the Network ' + chainId)
+  } else {
+    return baseUrl
+  }
+}
+
+async function _getErrorForBadPostOrderRequest(response: Response): Promise<string> {
+  let errorMessage: string
+  try {
+    const orderPostError: OrderPostError = await response.json()
+
+    switch (orderPostError.errorType) {
+      case 'DuplicateOrder':
+        errorMessage = 'There was another identical order already submitted'
+        break
+
+      case 'InsufficientFunds':
+        errorMessage = "The account doesn't have enough funds"
+        break
+
+      case 'InvalidSignature':
+        errorMessage = 'The order signature is invalid'
+        break
+
+      case 'MissingOrderData':
+        errorMessage = 'The order has missing information'
+        break
+
+      default:
+        console.error('Unknown reason for bad order submission', orderPostError)
+        errorMessage = orderPostError.description
+        break
+    }
+  } catch (error) {
+    console.error('Error handling a 400 error. Probably could not deserialize the JSON response')
+    errorMessage = 'The order was not accepted by the operator'
+  }
+
+  return errorMessage
+}
+
+async function _getErrorForUnsuccessfulPostOrder(response: Response): Promise<string> {
+  let errorMessage: string
+  switch (response.status) {
+    case 400:
+      errorMessage = await _getErrorForBadPostOrderRequest(response)
+      break
+
+    case 403:
+      errorMessage = 'The order cannot be accepted. Your account is deny-listed.'
+      break
+
+    case 429:
+      errorMessage = 'The order cannot be accepted. Too many order placements. Please, retry in a minute'
+      break
+
+    case 500:
+    default:
+      errorMessage = 'Error adding an order'
+  }
+  return errorMessage
+}
+
+export async function postSignedOrder(params: { chainId: ChainId; order: OrderCreation }): Promise<OrderID> {
+  const { chainId, order } = params
+  console.log('[utils:operator] Post signed order for network', chainId, order)
+
+  const orderRaw: Omit<OrderCreation, 'kind'> & { kind: string } = {
+    ...order,
+    // TODO: The NPM module will use the same structure as the API soon, this is temporal code too
+    kind: order.kind === ORDER_KIND_SELL ? 'sell' : 'buy'
+  }
+
+  // Call API
+  const baseUrl = _getApiBaseUrl(chainId)
+  const response = await fetch(`${baseUrl}/orders`, {
+    ...POST_HEADERS,
+    body: JSON.stringify(orderRaw)
+  })
+
+  // Handle respose
+  if (!response.ok) {
+    // Raise an exception
+    const errorMessage = await _getErrorForUnsuccessfulPostOrder(response)
+    throw new Error(errorMessage)
+  }
+
+  const uid = response.json()
```

Comment:
Isn't uuid not in json? `resopnse.text()` maybe? or at least `const uid = await response.json()` if it is indeed a prop in json